tools/transformer/src/utils.js

```diff
@@ -88,28 +106,119 @@ function detectDuplicates(objArray) {
   return duplicates;
 }
 
-function removeParentFromAllOf(child, parentName) {
-  child.allOf = child.allOf.filter((childAllOfItem) => {
-    if (childAllOfItem["$ref"]) {
-      if (
-        childAllOfItem["$ref"].endsWith("#/components/schemas/" + parentName)
-      ) {
-        return false;
-      }
+function expandReference(obj, apiObject) {
+  if (!obj || !obj["$ref"]) {
+    return obj;
+  }
+
+  const dereferencedObj = getObjectFromReference(obj, apiObject);
+  delete obj["$ref"];
+
+  Object.keys(dereferencedObj).forEach((key) => {
+    obj[key] = dereferencedObj[key];
+  });
+
+  return obj;
+}
+
+function flattenAllOfObject(obj, apiObject) {
```

Comment:
Btw. This is no longer utils. It is actual logic of AllOf transformation. 
Unless it is resused it should be probably outside utils.